### PR TITLE
Introduce matrix_norm in operator.assert_equal(), and circuit.assert_equal() and circuit.assert_equal_to_operator() call operator.assert_equal()

### PR DESCRIPTION
**Pull request by junnaka51**
Details are explained inline 🙇

### CODE REVIEW

File: quantestpy/circuit.py [hunk 1]
```diff
@@ -220,49 +222,6 @@ def _add_x_gate_in_front(gates: list, qubit: int) -> dict:
     raise QuantestPyAssertionError(msg)
 
 
-def _get_matrix_norm(
```
Comment (junnaka51) [inline quantestpy/circuit.py:223]: This function is moved to `quantestpy/operator.py` without any changes.

File: quantestpy/operator.py [hunk 2]
```diff
@@ -30,12 +30,56 @@ def assert_is_unitary(
         raise QuantestPyAssertionError(msg)
 
 
+def _get_matrix_norm(
```
Comment (junnaka51) [inline quantestpy/operator.py:33]: This function is moved from `quantestpy/operator.py` without any changes.

File: quantestpy/circuit.py [hunk 3]
```diff
@@ -25,6 +24,7 @@ def assert_equal_to_operator(
         rtol: float = 0.,
         atol: float = 1e-8,
         up_to_global_phase: bool = False,
+        matrix_norm_type: Union[str, None] = None,
```
Comment (junnaka51) [inline quantestpy/circuit.py:27]: `matrix_norm` is now introduced in circuit.assert_equal_to_operator()

File: quantestpy/circuit.py [hunk 4]
```diff
@@ -385,41 +344,13 @@ def assert_equal(
...
+    # call operator.assert_equal
+    operator.assert_equal(
+        whole_gates_a,
+        whole_gates_b,
+        rtol,
+        atol,
+        up_to_global_phase,
+        matrix_norm_type,
+        msg
+    )
```
Comment (junnaka51) [inline quantestpy/circuit.py:356]: Now `operator.assert_equal()` is called both in `circuit.assert_equal()` and in `circuit.assert_equal_to_operator()`, which enables us to maintain the package easily.

File: quantestpy/operator.py [hunk 5]
```diff
@@ -70,16 +114,47 @@ def assert_equal(
...
 
     # assert equal
-    try:
-        np.testing.assert_allclose(
-            actual=a,
-            desired=b,
-            rtol=rtol,
-            atol=atol,
-            err_msg=f"Up to global phase: {up_to_global_phase}"
+    if matrix_norm_type is None:
```
Comment (junnaka51) [inline quantestpy/operator.py:117]: This part is moved from `quantestpy/circuit.py`  with only small modifications.

File: test/test_operator_assert_equal.py [hunk 6]
```diff
@@ -1,7 +1,8 @@
 import unittest
 import numpy as np
+import traceback
 
-from quantestpy import operator
+from quantestpy.operator import assert_equal
```
Comment (junnaka51) [inline test/test_operator_assert_equal.py:5]: Directly import assert_equal method, which will be faster.

File: test/with_qiskit/test_circuit_assert_equal.py [hunk 7]
```diff
@@ -14,12 +14,12 @@ class TestCircuitAssertEqual(unittest.TestCase):
     How to execute this test:
```
Comment (junnaka51) [inline test/with_qiskit/test_circuit_assert_equal.py:14]: This is a test file. Only comments are updated.

File: test/with_qiskit/test_circuit_assert_equal.py [hunk 8]
```diff
@@ -14,12 +14,12 @@ class TestCircuitAssertEqual(unittest.TestCase):
...
     $ pwd
     {Your directory where you git-cloned quantestpy}/quantestpy
-    $ python -m unittest test.test_circuit_assert_equal
-    ...
+    $ python -m unittest test.with_qiskit.test_circuit_assert_equal
+    .s..........
```
Comment (mitz1012) [inline test/with_qiskit/test_circuit_assert_equal.py:18]: Why did you change line18?
What do you mean the new one?

File: test/with_qiskit/test_circuit_assert_equal.py [hunk 9]
```diff
@@ -14,12 +14,12 @@ class TestCircuitAssertEqual(unittest.TestCase):
...
     ----------------------------------------------------------------------
-    Ran 3 tests in 0.006s
+    Ran 12 tests in 0.012s
 
-    OK
+    OK (skipped=1)
```
Comment (mitz1012) [inline test/with_qiskit/test_circuit_assert_equal.py:22]: Doesn't `skipped=1` cause trouble?

**Review by mitz1012 — COMMENTED** [review]
I left my comments about [test/with_qiskit/test_circuit_assert_equal.py](https://github.com/QuantestPy/quantestpy/pull/124/files#diff-f64b816c54d04018318912ebe071ab7ba14e8b47782d35c9ff7295715b9c928e).

File: test/with_qiskit/test_circuit_assert_equal.py (hunk 8 above)
Comment (junnaka51) [reply test/with_qiskit/test_circuit_assert_equal.py:18]: One dot `.` means one test, therefore # of dots increase as # of tests increases.

File: test/with_qiskit/test_circuit_assert_equal.py (hunk 9 above)
Comment (junnaka51) [reply test/with_qiskit/test_circuit_assert_equal.py:22]: Thanks for this comment. One test is skipped on purpose (the reason was that rx or rz gate was not implemented at that time, or something like that). 
I will check it.

Comment (junnaka51) [reply test/with_qiskit/test_circuit_assert_equal.py]: @mitz1012 
> Doesn't `skipped=1` cause trouble?

I deleted this test because I have implemented this test without understanding correctly 👉 https://github.com/QuantestPy/quantestpy/pull/124/commits/22dfd4c16a5b5e915611fc7ab49fecbddd6e6af7

For details see https://github.com/QuantestPy/quantestpy/issues/129

Anyway, the skipping test problem that you have pointed out has been solved.

**Review by mitz1012 — APPROVED** [review]
LGTM!
